### DISCUSSION TITLE
Quick and dirty

### DISCUSSION THREAD

**Discussion by ghost** [LinusCDE/plato#14] [Ideas]
Hi, 

Would you be able to make a quick and dirty autoinstaller toggler like https://github.com/ddvk/remarkable-autoinstall/tree/master/rm2 for plato? 
I have no need for all the things toltec has to offer so a simple toggler would be enough for me.

**Comment by LinusCDE** [comment]
Hi,

Not sure how to tackle this. I have not used the autoinstaller, yet. It seems like a swap of koreader with plato might be possible. But I could be horribly wrong there. @ddvk has probably more knowledge here.

I used to have several little c programs with systemd services to map all button combinations with stopping xochitl and starting something else. This sadly won't really work on the rm 2 anymore.

You can run plato standalone by just untaring the [release file](https://github.com/LinusCDE/plato/releases/tag/0.9.17-rm-release-10) and running the .sh script. But some sort framebuffer shim is needed to display (like rm2fb), but the autoinstaller advises against it.

Some kind of launcher would make live certainly easier there. Toltec is not required for the launchers.

I can probably look more into it to the end of next week. Maybe @ddvk has some thoughts as well.

**Reply by ghost** [reply]
I've got some knowledge of linux and feel reasonable comfortable around the command promt, but I find it to become too technical for me when it comes to combining plato, a launcher and the framebuffer.  That's why I love ddvk's solution so much.  I was looking at plato because of an issue with koreader slow page turn after the rm2 has been sleeping for a while. ddvk is trying to reproduce it.

**Reply by LinusCDE** [reply]
That is a really interesting insight.

As already said, I'm not comfortable with ddvks hacks much apart (from usually having the xochitl hacks enabled). I personally also really like to know everything about a given system and toltec certainly makes this harder to grasp for the sake of distro-like package manager convenience.

The said reality on the rM 2 is that:
 - you can't just do a "dirt cheap launcher" anymore by listening for button combinations since those are missing
 - third party applications can't just use the eink display as a somewhat ordinary linux framebuffer anymore.

I personally am still a big proponent of keeping hacks standalone when possible. The convenience of toltect (at least for me) was that you don't need to figure out how to compile/install a mod yourself and keep track of it after fw upgrades. Many mods just have unified scripts to compile any given mod in the same environment and then also easily get them installed any managed.

ddvk is certainly doing a great thing by having a simple script be really effective in combination with his xochitl hacks and also stay fairly easy to understand. If you want plato in it, it is probably best to ask him to do that on his repo.

As already said, you can get a standalone version of plato on the release page (the `*-dist.zip`), extract it onto your device and run it. Only problem on the rm2 is, that you also need to setup the [remarkable2-framebuffer](https://github.com/ddvk/remarkable2-framebuffer/) stuff. The instructions on the readme should still be good to do this standalone as well.

Any kind of device mod (for example from [this great listing](https://github.com/reHackable/awesome-reMarkable)) requires some manual work. Many mods take a bit more work nowadays since many might have only targeted the rM 1 and have input problems or expect the physical buttons since a long time has passed and many devs are not interested anymore, have moved on or just don't care about a device they don't own since most projects probably were just passion hacking/porting for the device they own.

Toltec did a great job imo in getting interest in older projects active again and generally pushing more rM 2 support across the board so not every dev had to figure this out themselves.

---

P.S.: Sorry for the late resonse. Just didn't get to do it until now. I didn't proof read the text above so it might sound strange at some part or repeat itself or my previous answer.

**Reply by ghost** [reply]
Thank you for your detailed response.  For me, some minor improvements on the RM for reading epub ( tap to turn page ) would solve the need for other programs like plato, koreader or ddvk hacks. But I love the idea that it is possible to run other programs on the device if I needed to. In that case toltec makes the most sense.  It's totally a first world problem that in certain circumstances page turns get slow. So for now I'll just swipe out and into koreader and as long the device isn't going to sleep, the turns are swift. I just have to keep on reading, no breaks :-)

**Reply by LinusCDE** [reply]
Lol. For me the "Tap to switch page" is actually an annoying thing in plato. I really like xochil's approach since it means that you can touch the surface a lot without affecting anything. Swiping never annoyed me here.

When reading with plato I sometimes even just disable touch input to only tap with the pen.
